lib/kvdb/ikvdb.c

```diff
@@ -2595,17 +2594,68 @@ ikvdb_kvs_cursor_read(
             return cur->kc_err;
     }
 
-    err = kvs_cursor_read(cur, &kvt, eof);
+    err = kvs_cursor_read(cur, flags, eof);
     if (ev(err))
         return err;
     if (*eof)
         return 0;
 
-    *key = kvt.kvt_key.kt_data;
-    *key_len = kvt.kvt_key.kt_len;
+    *key = kvs_cursor_key_copy(cur, NULL, 0, key_len);
+    if (val) {
+        *val = kvs_cursor_val_copy(cur, NULL, 0, val_len);
+        if (ev(!val))
+            return merr(EBUG);
```

Comment:
Having a hard time understanding the EBUG usage here. It looks like one of the fallible operations in val_copy() is the decompress operation. Why not continue to carry the original error forward instead of masking it? And then there is a comparison with a min_t as well. To me it seems like EBUG is a misnomer, and anywhere we have EBUG should be an assert or an abort.

Reply:
Yes. Fixed it.